src/EprPrnIntegration.Common/Service/ServiceBusProvider.cs

```diff
@@ -1,45 +1,126 @@
 ﻿using Azure.Messaging.ServiceBus;
 using EprPrnIntegration.Common.Configuration;
 using EprPrnIntegration.Common.Models;
+using EprPrnIntegration.Common.Models.Queues;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using System.Text.Json;
 
 namespace EprPrnIntegration.Common.Service
 {
-    public class ServiceBusProvider : IServiceBusProvider
+    public class ServiceBusProvider(
+        ILogger<ServiceBusProvider> logger,
+        ServiceBusClient serviceBusClient,
+        IOptions<ServiceBusConfiguration> config) : IServiceBusProvider
     {
-        private readonly ILogger<ServiceBusProvider> _logger;
-        private readonly ServiceBusClient _serviceBusClient;
-        private readonly IOptions<ServiceBusConfiguration> _serviceBusConfig;
-        public ServiceBusProvider(ILogger<ServiceBusProvider> logger, ServiceBusClient serviceBusClient, IOptions<ServiceBusConfiguration> config)
-        {
-            _logger = logger;
-            _serviceBusClient = serviceBusClient;
-            _serviceBusConfig = config;
-        }
         public async Task SendFetchedNpwdPrnsToQueue(List<NpwdPrn> prns)
         {
             try
             {
-                await using var sender = _serviceBusClient.CreateSender(_serviceBusConfig.Value.FetchPrnQueueName);
+                await using var sender = serviceBusClient.CreateSender(config.Value.FetchPrnQueueName);
                 using ServiceBusMessageBatch messageBatch = await sender.CreateMessageBatchAsync();
                 foreach (var prn in prns)
                 {
                     var jsonPrn = JsonSerializer.Serialize(prn);
                     if (!messageBatch.TryAddMessage(new ServiceBusMessage(jsonPrn)))
                     {
-                        _logger.LogWarning("SendFetchedNpwdPrnsToQueue - The message with EvidenNo: {EvidenNo} is too large to fit in the batch.", prn.EvidenceNo);
+                        logger.LogWarning("SendFetchedNpwdPrnsToQueue - The message with EvidenNo: {EvidenNo} is too large to fit in the batch.", prn.EvidenceNo);
+                    }
+                }
+                await sender.SendMessagesAsync(messageBatch);
+                logger.LogInformation("SendFetchedNpwdPrnsToQueue - A batch of {MessageBatchCount} messages has been published to the queue: {queue}", messageBatch.Count, config.Value.FetchPrnQueueName);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("SendFetchedNpwdPrnsToQueue failed to add message on Queue with exception: {exception}", ex);
+                throw;
+            }
+        }
+
+        public async Task SendDeltaSyncExecutionToQueue(DeltaSyncExecution deltaSyncExecution)
+        {
+            try
+            {
+                await using var sender = serviceBusClient.CreateSender(config.Value.DeltaSyncQueueName);
+                using var messageBatch = await sender.CreateMessageBatchAsync();
```

Comment:
I don't think we are going to have batch here we are pushing only one message so no need of batch which will save additional logic around try add